### PR TITLE
Silence warnings & NA value behavior

### PR DESCRIPTION
**Pull request by kirstyward**
1. Upgrade altair to 5.1.2 - this gets rid of FutureWarnings for deprecated pandas functions
2. Add check for NA values in calculate_votes()
3. Catch threshold = zscore in vote calculation
4. Remove deprecated na inf pandas option
5. Add check to make sure votes df has only valid values in calculate_summary()

### CODE REVIEW

**Comment by kirstyward** [comment]
@pmbaumgartner  Branch is ready for you again. The calculate_votes() function now preserves NA values, and visuals calculate percentages without NA's included

File: tests/test_find_votes.py [hunk 1]
```diff
@@ -59,6 +60,18 @@ def test_provide_threshold(sample_data):
     calculate_votes(data=sample_data[0], columns=sample_data[1], threshold=0.5)
 
 
+def test_na_column(sample_data):
+    nan_data = sample_data[0].assign(Q6=np.NaN)
+    calculate_votes(data=nan_data, columns=[f"Q{i}" for i in range(1, 7)])
```
Comment (pmbaumgartner) [inline tests/test_find_votes.py:65]: Is there an assertion we want to make with this result? Maybe that the final result is not `NaN`?

File: tests/test_find_votes.py [hunk 2]
```diff
@@ -59,6 +60,18 @@ def test_provide_threshold(sample_data):
...
+
+
+def test_missing_data(sample_data):
+    data = sample_data[0]
+    data["Q6"] = [1, 2, 3, 4, np.NaN, 4, 4, 2, 1, 5]
+
+    votes, stats = calculate_votes(data, columns=[f"Q{i}" for i in range(1, 7)])
```
Comment (pmbaumgartner) [inline tests/test_find_votes.py:72]: Same idea here - what is true about this data that we would want to assert?

Comment (kirstyward) [reply tests/test_find_votes.py:72]: Updated this - the test now checks that the NA is preserved, not assigned as a vote

File: tests/test_find_votes.py (hunk 1 above)
Comment (kirstyward) [reply tests/test_find_votes.py:65]: I don't think this test is needed, going to remove

**Pull request merged**